internal/pkg/pdkshell/pdkshell.go

```diff
@@ -0,0 +1,91 @@
+package pdkshell
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/rs/zerolog/log"
+)
+
+type PDKInfo struct {
+	RubyVersion      string
+	InstallDirectory string
+	RubyExecutable   string
+	PDKExecutable    string
+	CertDirectory    string
+	CertPemFile      string
+}
+
+// Execute runs a given pdk command
+// It first detects where the PDK Ruby installation is on the local system
+// Then it builds the correct command line to execute the PDK ruby with provided arguments
+func Execute(args []string) (int, error) {
+	i := getPDKInfo()
+	executable := buildExecutable(i.RubyExecutable)
+	args = buildCommandArgs(args, i.RubyExecutable, i.PDKExecutable)
+	env := os.Environ()
+	env = append(env, fmt.Sprintf("SSL_CERT_DIR=%s", i.CertDirectory), fmt.Sprintf("SSL_CERT_FILE=%s", i.CertPemFile))
+	cmd := &exec.Cmd{
+		Path:   executable,
+		Args:   args,
+		Stdout: os.Stdout,
+		Stderr: os.Stderr,
+		Env:    env,
+	}
+
+	log.Trace().Msgf("args: %s", args)
+	if err := cmd.Run(); err != nil {
+		log.Fatal().Msgf("pdk failed with '%s'\n", err)
+		return 1, err
+	}
+
+	return 0, nil
+}
+
+// getPDKInfo detects where the PDK Ruby installation is on the local file system
+// It handles detecting the installation on Windows and other platforms
+func getPDKInfo() *PDKInfo {
+	rubyVersion := "2.4.10"
```

Comment:
Until we move away from PDK as a gem inside one specific version of Ruby, we won't be able to avoid having a hardcoded Ruby version. 

Existing platform implementations hardcoding ruby version:

[windows](https://github.com/puppetlabs/pdk-vanagon/blob/86aa40cb511ffdd81c4343af0996d4e1ad3dc501/resources/files/windows/PuppetDevelopmentKit/PuppetDevelopmentKit.psm1#L6)

[*nix](https://github.com/puppetlabs/pdk-vanagon/blob/86aa40cb511ffdd81c4343af0996d4e1ad3dc501/resources/files/posix/pdk_env_wrapper#L4)